### PR TITLE
HPCC-25577 Capture extra info. when regression query times out

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
Coincidentally this PR hit the issue it's trying to diagnose, and although there was a issue in the smoketest-preabort.sh script (didn't capture stacks), it did capture the process list and it reveals :

```
2021-03-22T12:54:49.6736165Z [Failure] 184. Timeout occured for dfse2.ecl and no more attempt left. Force to abort... 
..
2021-03-22T12:54:45.1427236Z [Failure] 184. Create Stack Trace for roxie (pid:9641) into '/home/runner/work/HPCC-Platform/HPCC-Platform/regress/log/W20210322-124245-roxie.trace'
2021-03-22T12:54:49.4730238Z [Failure] 184. Execute pre abort script 'smoketest-preabort.sh'
2021-03-22T12:54:49.6353293Z [Failure] 184. Pre abort script result stored into '/home/runner/work/HPCC-Platform/HPCC-Platform/regress/log/W20210322-124245-preAbort.log'
2021-03-22T12:54:49.6736165Z [Failure] 184. Timeout occured for dfse2.ecl and no more attempt left. Force to abort... 
```
runner     24050  0.0  1.1 4031756 85228 ?       Sl   12:42   0:00 /home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/bin/eclcc -shared - --timings --xml --nostdinc --metacache= --logfile=W20210322-124245.eclcc.log -dfs=10.1.0.4:7070 -scope=regress -cluster=roxie -wuid=W20210322-124245 -token=HPCC[u=,w=W20210322-124245]; -fmaxCompileThreads=2 -fpassDFSToEclcc=1 -oW20210322-124245 -platform=roxie --component=myeclccserver@10.1.0.4 -fcreated_by=ws_workunits -fdisablelocaloptimizations=1 -fallowvariableroxiefilenames=1 -ffailonleaks=1 -fcreated_for=regress -fpickbestengine=false -Dhpccbasedir='/home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/'
```

job started @ 12:42:
```
2021-03-22T12:42:45.1623914Z [Action] 184. Test: dfse2.ecl
```

so it appears eclcc has stalled/deadlocked on something.

**Comment by jakesmith** [comment]
I've now updated the script to also capture stacks for eclcc..

File: .github/workflows/smoketest-preabort.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+echo 'List of processes:'
+ps aux 
+echo 'esp stacks'
+echo '=========='
+gdb --batch --quiet -ex "set interactive-mode off" -ex "thread apply all bt" -ex "quit" esp `pidof esp`
```
Comment (AttilaVamos) [inline .github/workflows/smoketest-preabort.sh:5]: It seems something wrong with "esp `pidof esp`".
In the regress\log\W20210322-124245-preAbort.log file: 
```
esp stacks
==========
/home/jsmith/hpccDebug/opt/HPCCSystems/bin/esp: No such file or directory.
Could not attach to process.  If your uid matches the uid of the target
process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try
again as the root user.  For more details, see /etc/sysctl.d/10-ptrace.conf
ptrace: Operation not permitted.
/home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/testing/regress/9414: No such file or directory. 
```
Same problem with other stack traces.

Comment (jakesmith) [reply .github/workflows/smoketest-preabort.sh:5]: yes, it was fixed shortly after (removed erroneous path to exes)

Comment (AttilaVamos) [reply .github/workflows/smoketest-preabort.sh:5]: Should we re-run smoketest action again to check what it can catch if dfse2 problem occurred again?

**Review by AttilaVamos — APPROVED** [review]

**Comment by jakesmith** [comment]
I'll close this PR for now, I have reproduced the issue on a separate PR branch, so there's no need necessarily to merge it now, to spot this particular problem..
But still useful to capture similar issue for other future cases, so will re-open with a more general script at some point.

**Pull request closed without merging**